examples/research_projects/codeparrot/scripts/human_eval.py

```diff
@@ -51,14 +51,23 @@ def main():
     # Load model and tokenizer
     tokenizer = AutoTokenizer.from_pretrained(args.model_ckpt)
     model = AutoModelForCausalLM.from_pretrained(args.model_ckpt)
-    pipe = pipeline("text-generation", model=model, tokenizer=tokenizer, device=-1)
+    pipe = pipeline("text-generation", model=model, tokenizer=tokenizer, device=args.device_int)
 
     # Load evaluation dataset and metric
     human_eval = load_dataset("openai_humaneval")
     code_eval_metric = load_metric("code_eval")
 
+    # Run a quick test to see if code evaluation is enabled
+    try:
+        _ = code_eval_metric.compute(references=[""], predictions=[[""]])
+    except Exception as e:
+        print(
+            'Code evaluation not enabled. Read the warning below carefully and then use `--HF_ALLOW_CODE_EVAL="1"` flag to enable code evaluation.'
+        )
+        raise e
```

Comment:
It would be good to be more concrete about the exception. If there is another issue (e.g. on Windows this will throw a `NotImplementedError`) the user would be very confused.

I should have thought a bit more about my previous suggestion!

```suggestion
    except ValueError as exception:
        print(
            'Code evaluation not enabled. Read the warning below carefully and then use `--HF_ALLOW_CODE_EVAL="1"` flag to enable code evaluation.'
        )
        raise exception
```

Reply:
Oooooo i didn't realize that was a thing, I'll need to remember this in future!

Reply:
I just noticed that you undid  the fix of `self.tokenizer` fix in the `ConstantLengthDataset`. Could you add it back?

Reply:
> I just noticed that you undid the fix of `self.tokenizer` in the `ConstantLengthDataset`. Could you add it back?

On it! Forgot about that too haha